server/schema/resolvers.js

```diff
@@ -3,63 +3,70 @@ const { User, Supplier } = require('../models');
+            return Supplier.find();
+        },
+        supplier: async (parent, { supplierId }) => {
+            return Supplier.findOne({ _id: supplierId });
+        },
+        me: async (parent, args, context) => {
+            if (context.user) {
+                return User.findOne({ _id: context.user._id });
+            }
+            throw new AuthenticationError('You need to be logged in!');
+        },
     },
-    suppliers: async () => {
-        return Supplier.find();
-    },
-    supplier: async (parent, { supplierId }) => {
-      return Supplier.findOne({ _id: supplierId });
-    },
-    me: async (parent, args, context) => {
-      if (context.user) {
-        return User.findOne({ _id: context.user._id });
-      }
-      throw new AuthenticationError('You need to be logged in!');
-    },
-  },
 
-  Mutation: {
-    addUser: async (parent, { username, email, password }) => {
-      const user = await User.create({ username, email, password });
-      const token = signToken(user);
-      return { token, user };
-    },
-    login: async (parent, { email, password }) => {
-      const user = await User.findOne({ email });
+    Mutation: {
+        addUser: async (parent, { username, email, password }) => {
+            const user = await User.create({ username, email, password });
+            const token = signToken(user);
+            return { token, user };
+        },
+        login: async (parent, { email, password }) => {
+            const user = await User.findOne({ email });
 
-      if (!user) {
-        throw new AuthenticationError('No user found with this email address');
-      }
+            if (!user) {
+                throw new AuthenticationError('No user found with this email address');
+            }
 
-      const correctPw = await user.isCorrectPassword(password);
+            const correctPw = await user.isCorrectPassword(password);
 
-      if (!correctPw) {
-        throw new AuthenticationError('Incorrect credentials');
-      }
+            if (!correctPw) {
+                throw new AuthenticationError('Incorrect credentials');
+            }
 
-      const token = signToken(user);
+            const token = signToken(user);
 
-      return { token, user };
-    },
-    addSupplier: async (parent, args, context) => {
-        
-    },
-    updateSupplier: async (parent, args, context) => {
-      
-    },
-    deleteSupplier: async (parent, { _id }, context) => {
-      
-    },
-    updateUserMaterial: async (parent, args, context) => {
-      
-    },
-    updateSupplierMaterial: async (parent, args, context) => {
-      
+            return { token, user };
+        },
+        addSupplier: async (parent, { supplierText }, context) => {
+            if (context.user) {
+                const newSupplier = Supplier.create({
+                    supplierText,
+                    user: context.user._id,
+                });
+                return newSupplier;
+            }
```

Comment:
I think we need to specify the arguments for the mutation as: 
```suggestion
        addSupplier: async (parent, { name, email, address, phone }, context) => {
            if (context.user) {
                const newSupplier = Supplier.create({
                    name, 
                    email, 
                    address,
                    phone,
                    user: context.user._id,
                });
                return newSupplier;
            }
```